src/lib/metrics.ts

```diff
@@ -694,6 +700,16 @@ export default class MetricsMonitor {
             mapFeaturesForClientDuration.observe(duration);
         });
 
+        events.onMetricEvent(
+            eventBus,
+            events.REQUEST_ORIGIN,
+            ({ type, method }) => {
+                if (flagResolver.isEnabled('originMiddleware')) {
+                    requestOriginCounter.increment({ type, method });
+                }
+            },
+        );
```

Comment:
This could later be simplified to:

```ts
events.onMetricEvent(eventBus, events.REQUEST_ORIGIN, requestOriginCounter.increment)
``` 

Which would still satisfy our types.